contrib/runners/local_runner/local_runner/runner.yaml

```diff
@@ -35,6 +35,31 @@
       description: Action timeout in seconds. Action will get killed if it doesn't
         finish in timeout seconds.
       type: integer
+  output_schema:
+    succeeded:
+      type: boolean
+    failed:
+      type: boolean
+    return_code:
+      type: integer
+    stderr:
+      anyOf:
```

Comment:
Hm, `stderr` and `stdout` should always be strings, right? Or am I missing something?

Reply:
You would think, but the way our jsonify works is it traverses any keys and tries to render JSON. So any valid JSON is rendered. For instances if `stdout` is `123` you get the functional equivalent of this:

```
In [1]: import json

In [2]: type(json.loads('123'))
Out[2]: int
```

Reply:
Ah, that's a bit nasty, I thought this only happens for result not also for stdout and stderr.

Reply:
EDIT: Had a look again and notice that's just for the local runner, Python runner is "fine".

Still a bit nasty, but yeah, sadly an existing behavior.

Reply:
If I'm reading this correctly it's all output for every runner.
https://github.com/StackStorm/st2/blob/fd9c6f59887efe588ecbce562ac0d80b491edd6a/st2actions/st2actions/container/base.py#L129

Reply:
Doesn't seem so for Python runner:

```bash
  stderr: '4321'
  stdout: '1234'

kami ~/w/stackstorm/st2 (git:master)$ cat contrib/examples/actions/pythonactions/print_python_version.py
import sys

from st2common.runners.base_action import Action


class PrintPythonVersionAction(Action):

    def run(self):
        sys.stdout.write('1234')
        sys.stderr.write('4321')
```

And looking and `try_json` source, it doesn't seem to do any type casting.